.github/workflows/nm-build-test.yml

```diff
@@ -173,7 +176,7 @@ jobs:
 
     # update docker
     DOCKER:
-        needs: [BUILD, UPLOAD]
+        needs: [TEST, BENCHMARK, LM-EVAL]
```

Comment:
Should we keep the old needs here?

Reply:
i think we should have the same criteria for the docker. the issues i had with the previous list was that "BUILD" was redundant and "UPLOAD" isn't the same as "TEST, BENCHMARK, LM-EVAL". an example would be upload failing due to auth or resource limits.